src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -63,8 +28,44 @@
   {
     return ImmutableMap.copyOf(variableVertices);
   }
+  public VariableReferenceVertex getVariableVertex(VariableIdentifier vID)
+      throws VariableNotDefinedException {
+    if (variableVertices.containsKey(vID)) {
+      return variableVertices.get(vID);
+    } else {
+      throw new VariableNotDefinedException(vID);
+    }
+  }
+
+  public void createVariableVertex(VariableIdentifier vID)
+      throws MultipleDefinitionException {
+    if (variableVertices.containsKey(vID)) {
+      throw new MultipleDefinitionException(vID);
+    } else {
+      variableVertices.put(vID, new VariableReferenceVertex(this, vID));
+    }
+  }
+
+  private List<ExpressionVertex> nonVariableVertices = new LinkedList<>();
+  public List<ExpressionVertex> getNonVariableVertices() {
+    return ImmutableList.copyOf(nonVariableVertices);
+  }
+  public void addNonVariableVertex(ExpressionVertex v) {
```

Comment:
Maybe just overload so that the api is like `addVertex(VariableIdentifier)` `addVertex(ExpressionVertex)`